debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleOffsetContext.java

```diff
@@ -23,33 +25,55 @@
 public class OracleOffsetContext implements OffsetContext {
 
     public static final String SNAPSHOT_COMPLETED_KEY = "snapshot_completed";
+    public static final String SNAPSHOT_PENDING_TRANSACTIONS_KEY = "snapshot_pending_tx";
+    public static final String SNAPSHOT_SCN_KEY = "snapshot_scn";
 
     private final Schema sourceInfoSchema;
 
     private final SourceInfo sourceInfo;
     private final TransactionContext transactionContext;
     private final IncrementalSnapshotContext<TableId> incrementalSnapshotContext;
 
+    /**
+     * SCN that was used for the initial consistent snapshot.
+     *
+     * We keep track of this field because it's a cutoff for emitting DDL statements,
+     * in case we start mining _before_ the snapshot SCN to cover transactions that were 
+     * ongoing at the time the snapshot was taken.
+     */
+    private final Scn snapshotScn;
+
+    /**
+     * Map of (txid, start SCN) for all transactions in progress at the time the
+     * snapshot was taken.
+     */
+    private Map<String, Scn> snapshotPendingTransactions;
+
     /**
      * Whether a snapshot has been completed or not.
      */
     private boolean snapshotCompleted;
 
     public OracleOffsetContext(OracleConnectorConfig connectorConfig, Scn scn, Scn commitScn, String lcrPosition,
+                               Scn snapshotScn, Map<String, Scn> snapshotPendingTransactions,
```

Comment:
This change leads to a compilation problem with `XStreamOracleOffsetContextLoader`.  I suspect we probably need these lines added before the constructor & the arguments passed:

```
        final Map<String, Scn> snapshotPendingTransactions = OracleOffsetContext.loadSnapshotPendingTransactions(offset);
        final Scn snapshotScn = OracleOffsetContext.loadSnapshotScn(offset);
        return new OracleOffsetContext(connectorConfig, scn, lcrPosition, snapshotScn, snapshotPendingTransactions,
                snapshot, snapshotCompleted, TransactionContext.load(offset), SignalBasedIncrementalSnapshotContext.load(offset));
```

Reply:
Also if this is only applicable for Xstream and not LogMiner, then the tests should be annotated to be run only when the adapter is LogMiner and not Xstream.